test/silicon_nlcc.jl

```diff
@@ -1,7 +1,9 @@
-include("run_scf_and_compare.jl")
-include("testcases.jl")
+@testsetup module SiliconNLCC
+using DFTK
+using LazyArtifacts
 
-function run_silicon_nlcc(T; Ecut=5, grid_size=15, spin_polarization=:none, kwargs...)
+function run_silicon_nlcc(T, silicon, run_scf_and_compare; Ecut=5, grid_size=15,
```

Comment:
Maybe rename to `testcase` for consistency with other functions with a variable carrying an unspecified testcase.